### ISSUE TITLE
ng test, passed on macOS (it should not) but fails on CI and also fails if i run npx jest

### ISSUE THREAD

**Issue by iangregsondev** [nrwl/nx#1997]
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository (not related to Angular, AngularCLI or any dependency)

## Expected Behavior

When running jest via the command line - i get a failure, its because an internal class is throwing process.exit(1) - i know why this is happening BUT the problem is then my mac it throws zero errors but the CI that runs the same job, hightlights the issue...

This is from CI

```
Pattern:  - 0 matches
  ●  process.exit called with "1"



      at ConfigService.performValidationChecks (src/lib/config.service.ts:2754:15)
      at new ConfigService (src/lib/config.service.ts:2427:10)
      at Injector.instantiateClass (../../../../node_modules/@nestjs/core/injector/injector.js:268:19)
      at callback (../../../../node_modules/@nestjs/core/injector/injector.js:74:41)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! api@0.0.0-development test: `ng test "--code-coverage"`
npm ERR! Exit status 1
npm ERR! 
```

Here is my mac version

```
 PASS  libs/api/common/config/src/lib/config.service.spec.ts
  ConfigService
    ✓ should be defined (14ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        2.732s, estimated 3s
```

and running jest on the command line also hightlights the issue but above the issue is not highighted

```
 PASS  apps/api/src/app/app.controller.spec.ts
  ●  process.exit called with "1"

      144 |
      145 |     if (fatalError) {
    > 146 |       process.exit(1)
          |               ^
      147 |     }
      148 |   }
      149 |

      at ConfigService.performValidationChecks (libs/api/common/config/src/lib/config.service.ts:146:15)
      at new ConfigService (libs/api/common/config/src/lib/config.service.ts:97:10)
      at Injector.instantiateClass (node_modules/@nestjs/core/injector/injector.js:268:19)
      at callback (node_modules/@nestjs/core/injector/injector.js:74:41)
  ●  process.exit called with "1"

      144 |
      145 |     if (fatalError) {
    > 146 |       process.exit(1)
          |               ^
      147 |     }
      148 |   }
      149 |

      at ConfigService.performValidationChecks (libs/api/common/config/src/lib/config.service.ts:146:15)
      at new ConfigService (libs/api/common/config/src/lib/config.service.ts:97:10)
      at Injector.instantiateClass (node_modules/@nestjs/core/injector/injector.js:268:19)
      at callback (node_modules/@nestjs/core/injector/injector.js:74:41)
  ●  process.exit called with "1"

      144 |
      145 |     if (fatalError) {
    > 146 |       process.exit(1)
          |               ^
      147 |     }
      148 |   }
      149 |

      at ConfigService.performValidationChecks (libs/api/common/config/src/lib/config.service.ts:146:15)
      at new ConfigService (libs/api/common/config/src/lib/config.service.ts:97:10)
      at Injector.instantiateClass (node_modules/@nestjs/core/injector/injector.js:268:19)
      at callback (node_modules/@nestjs/core/injector/injector.js:74:41)
  ●  process.exit called with "1"

      144 |
      145 |     if (fatalError) {
    > 146 |       process.exit(1)
          |               ^
      147 |     }
      148 |   }
      149 |

      at ConfigService.performValidationChecks (libs/api/common/config/src/lib/config.service.ts:146:15)
      at new ConfigService (libs/api/common/config/src/lib/config.service.ts:97:10)
      at Injector.instantiateClass (node_modules/@nestjs/core/injector/injector.js:268:19)
      at callback (node_modules/@nestjs/core/injector/injector.js:74:41)
 FAIL  libs/api/common/config/src/lib/config.service.spec.ts
```

## Failure Information (for bugs)

So to summarise, CI thats runs the same code and the same command "npm run test --code-coverage" fails.

Running JEST on my mac i.e npx jest, also fails.

but running npm run test --code-coverage, passes on the mac.

The failure is expected, i mean I see an error in the mocking of the test but what worries me is that in the macos running the test script - its not hightlighted.

Can you help ?

**Comment by iangregsondev** [comment]
my library calls process.exit becasue i have some missing env variables.. which is correct, i forgot to mock them BUT the problem isn't highlighted in ng test

Is it because of 

  passWithNoTests: true

**Comment by FrozenPandaz** [comment]
I am unable to reproduce the issue even with `passWithNoTests: true`.

![image](https://user-images.githubusercontent.com/8104246/68056087-a7c6e780-fcc8-11e9-8f57-a0fc19ac52d9.png)


Please provide a repo to aid our investigation into this issue.

**Comment by iangregsondev** [comment]
It only seems to happen in my rather large (company) repo. I can't upload that. I will close it now and if I get any insight to it later then i will share.
